### PR TITLE
Incremental 02c - Unit tests for DeepCopy : Device to Host

### PR DESCRIPTION
**Pull request by rgayatri23**

### CODE REVIEW

**Comment by jeffmiles63** [comment]
see comments in #2536

File: core/unit_test/incremental_testing/Test02c_MemorySpace.hpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    dataType *DeviceData = (dataType*) Kokkos::kokkos_malloc<MemSpaceD>("DeviceData", num_elements*sizeof(dataType));
+    ASSERT_FALSE(DeviceData == nullptr);
+
+    //Allocate memory on Host space
+    dataType *HostData = (dataType*) Kokkos::kokkos_malloc<MemSpaceH>("HostData", num_elements*sizeof(dataType));
+    ASSERT_FALSE(HostData == nullptr);
+
+    for(int i = 0; i < num_elements; ++i)
+    {
+      DeviceData[i] = value;
```
Comment (dalg24) [inline core/unit_test/incremental_testing/Test02c_MemorySpace.hpp]: Can't write to the device from the host.

File: core/unit_test/incremental_testing/Test02c_MemorySpace.hpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        "HostData", num_elements * sizeof(dataType));
+    ASSERT_FALSE(HostData_send == nullptr);
+
+    // Allocate memory on Host space
+    dataType *HostData_recv = (dataType *)Kokkos::kokkos_malloc<MemSpaceH>(
+        "HostData", num_elements * sizeof(dataType));
+    ASSERT_FALSE(HostData_recv == nullptr);
+
+    for (int i = 0; i < num_elements; ++i) {
+      HostData_send[i] = value;
```
Comment (crtrott) [inline core/unit_test/incremental_testing/Test02c_MemorySpace.hpp]: fill there with different values for each element (something like `value + i` or so).

**Comment by rgayatri23** [comment]
All the requested changes have been addressed.

**Comment by rgayatri23** [comment]
Retest this please

File: core/unit_test/incremental/Test02c_MemorySpace.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        "HostData", num_elements * sizeof(dataType));
+    ASSERT_FALSE(hostData_send == nullptr);
+
+    // Allocate memory on Host space
+    dataType *hostData_recv = (dataType *)Kokkos::kokkos_malloc<memSpaceH>(
+        "HostData", num_elements * sizeof(dataType));
+    ASSERT_FALSE(hostData_recv == nullptr);
+
+    for (int i = 0; i < num_elements; ++i) {
+      hostData_send[i] = value;
```
Comment (crtrott) [inline core/unit_test/incremental/Test02c_MemorySpace.hpp]: this still needs to be changed to something like:
```
hostData_send[i] = value + dataType(i);
```

**Comment by crtrott** [comment]
closed in favour of #2633

**Pull request closed without merging**